app/routes/api/v1/warehouse/orders/dates/within-delivered/index.ts

```diff
@@ -0,0 +1,49 @@
+import { ActionFunction } from "@remix-run/node";
+import { z } from "zod";
+import {
+  badRequest,
+  methodNotAllowed,
+  notFoundRequest,
+} from "~/helpers/app-helpers.server";
+import { parseBody } from "~/lib/parse-body.server";
+import { findOrdersWithinDeliveredDates } from "~/models/order/order.server";
+
+export const action: ActionFunction = async ({ request }) => {
+  if (request.method.toLowerCase() != "post") {
+    methodNotAllowed();
+  }
+
+  const schema = z.object({
+    startDate: z.nullable(z.date()),
+    endDate: z.nullable(z.date()),
+  });
+
+  const parsedData = await parseBody(request, schema);
+
+  if (parsedData.success) {
+    const data = parsedData.data;
+
```

Comment:
`parseBody()` returns the data already checked, no need to do `if (parsedData.success)`